string-manipulation-1/src/get-words.js

```diff
@@ -0,0 +1,51 @@
+/* exported getWords */
+function getWords(string) {
+  var words = [];
+  var currentIdx = 0;
+  var i = 0;
+  var j = 0;
+  var oneWord = '';
+  var finalWord = '';
+
+  // debugger;
+
+  if (string === '') {
+    return words;
+  }
+
+  var spaces = [];
+  for (var h = 0; h < string.length; h++) {
+    if (string[h] === ' ') {
+      spaces.push(string[h]);
+      break;
+    }
+  }
+  if (spaces.length === 0) {
+    for (var z = 0; z < string.length; z++) {
+      oneWord += string[z];
+    }
+    words.push(oneWord);
+    return words;
+  }
+
+  while (i < string.length) {
+    if (string[i] === ' ') {
+      var word = '';
+      while (currentIdx < i) {
+        if (string[currentIdx] !== ' ') {
+          word += string[currentIdx];
+        }
+        currentIdx++;
+      }
+      words.push(word);
+    }
+    i++;
+  }
+
+  for (j = currentIdx + 1; j < string.length; j++) {
+    finalWord += string[j];
+  }
+  words.push(finalWord);
+
+  return words;
```

Comment:
There is a lot going on here, I think this can be simplified by a lot. If you loop over your string from start to end, here is the logical flow you want to follow:
1. Create a variable with an initial value of an empty string to hold the current word.
2. Create a variable holding an empty array for all the letters
3. Loop over your string one character at a time
     - If the character is not a space, add it onto your current word you are building
     - If the character is a space, push the current value fo your word variable to your list of words, then clear the value of the word variable back to an empty string
4. Once the loop has ended, push the final value of the word variable into your list of words (the last word won't have a space after it, but the variable will hold that full word if you let the whole loop finish)